### DISCUSSION TITLE
animation plugin produces a delay before showing first stimulus

### DISCUSSION THREAD

**Discussion by becky-gilbert** [jspsych/jsPsych#782] [Q&A] [answered]
Hi all,

The animation plugin seems to produce a delay before the first image is shown. I created a little demo here: https://xukotxedw5.cognition.run/
In the demo, when an image is shown via the animation plugin, there's a blank screen between the previous trial (fixation) and first image. But there's no delay between fixation and image when the image is shown via the image-keyboard-response plugin. 

I think this has to do with the fact that the [code that manipulates the display is inside of a setInterval function](https://github.com/jspsych/jsPsych/blob/83980085ef604c815f0d97ab55c816219e969b84/plugins/jspsych-animation.js#L69), so when the trial starts, the first stim isn't shown until _after_ the interval duration (frame_time + frame_isi) has passed. 

I think this could be easily fixed by changing the plugin so that the first image is shown immediately, and then setInterval could be used to control the rest of the display changes. Just wanted to get some input here before suggesting changes to the plugin file. Maybe I'm doing something wrong when setting up the trial, or maybe this is a feature rather than a bug?

Cheers,
Becky

**Comment by vijaymarupudi** [comment, marked as answer]
Atleast from reading the documentation, this seems like a bug to me, and I think your fix is good!

**Reply by becky-gilbert** [reply]
Ok, thanks for the feedback @vijaymarupudi!

**Comment by yuvalharr** [comment]
Hey There,

Thanks for this! I was having trouble with this as well.

Any update to the code yet?
If not, are you willing to share the fix you did in the plugin? I'm pretty new to js and not sure how to edit this..

Thanks!

**Reply by becky-gilbert** [reply]
Hi @yuvalharr, sorry for the slow response - I somehow missed this question.
Have you managed to get this working in the meantime? If not, I'll edit the code and post here as soon as I get the chance.

**Comment by jodeleeuw** [comment]
In case anyone is finding this after the fact, this issue has been fixed as of the 1.0.0 release of the plugin-animation package.
